### PR TITLE
Move the platform independent implementation to the common engine

### PR DESCRIPTION
**Pull request by kkristof**
Signed-off-by: Kristof Kosztyo <kkristof@inf.u-szeged.hu>

### CODE REVIEW

**Comment by kkristof** [comment]
Well, the bot isn't happy since the vulkan backend is a bit void at the moment :(

**Comment by kkristof** [comment]
To make the bot happy the #58 should be landed first.

File: src/gepard-engine.cpp [hunk 1]
```diff
@@ -27,4 +27,189 @@
 
 namespace gepard {
 
+/*!
+ * \brief GepardEngine::closePath
+ */
+void GepardEngine::closePath()
+{
+    _engineBackend->closePath();
```
Comment (szledan) [inline src/gepard-engine.cpp]: Please check the `_engineBackend`.
```c++
if (_engineBackend)
    _engineBackend->closePath();
```

File: src/gepard-engine.cpp [hunk 2]
```diff
@@ -27,4 +27,189 @@
...
+ * \param x  X-axis value of _start_ and _end_ point
+ * \param y  Y-axis value of _start_ and _end_ point
+ * \param w  size on X-axis
+ * \param h  size on Y-axis
+ *
+ * \todo documentation is missing
+ */
+void GepardEngine::fillRect(float x, float y, float w, float h)
+{
+    _engineBackend->fillRect(x, y, w, h);
```
Comment (szledan) [inline src/gepard-engine.cpp]: Please check the `_engineBackend`.
```c++
if (_engineBackend) {
    _engineBackend->fillRect(x, y, w, h);
}
```

**Comment by dbatyai** [comment]
LGTM

**Comment by szledan** [comment]
LGTM

**Pull request merged**